src/js/disability-benefits/526EZ/config/form.js

```diff
@@ -296,9 +325,212 @@ const formConfig = {
+                    }
+                  }
+                }
+              }
+            }
+          },
+          schema: {
+            type: 'object',
+            properties: {
+              disabilities: {
+                type: 'array',
+                items: {
+                  type: 'object',
+                  properties: {
+                    treatments: {
+                      type: 'array',
+                      items: {
+                        type: 'object',
+                        properties: {
+                          treatment: {
+                            type: 'object',
+                            properties: _.omit(['treatmentCenterType'], privateMedicalRecordsReleaseTreatmentSchema)
+                          }
+                        }
+                      }
+                    }
+                  }
+                }
+              }
+            }
+          }
+        },
+        // TODO: should this be renamed recordUpload?
+        documentUpload: {
+          title: 'Upload your private medical records',
+          depends: (formData, index) => {
+            const hasRecords = _.get(`disabilities.${index}.view:privateMedicalRecords`, formData);
+            // TODO: enable once previous chapter merged
+            // const uploadRecords = _.get(`disabilities.${index}.view:uploadPrivateRecords`, formData);
+            const uploadRecords = true;
+            return hasRecords && uploadRecords;
+          },
+          path: 'supporting-evidence/:index/documents',
+          // editPageOnReview: true,
+          showPagePerItem: true,
+          arrayPath: 'disabilities',
+          uiSchema: {
+            disabilities: {
+              items: {
+                medicalRecords: Object.assign({},
+                  fileUploadUI('Upload your private medical records', {
+                    allowRename: true,
+                    itemDescription: 'Adding additional evidence:',
+                    endpoint: '/v0/preneeds/preneed_attachments', // TODO: update this with correct endpoint (e.g. '/v0/21-526EZ/medical_records')
+                    fileTypes: ['pdf', 'jpg', 'jpeg', 'tiff', 'tif', 'png'],
+                    maxSize: 27262976, // TODO: create constant
+                    createPayload: (file) => {
+                      const payload = new FormData();
+                      payload.append('preneed_attachment[file_data]', file); // TODO: update this with correct property (e.g. 'health_record[file_data]')
+
+                      return payload;
+                    },
+                    parseResponse: (response, file) => {
+                      return {
+                        name: file.name,
+                        confirmationCode: response.data.attributes.guid
+                      };
+                    },
+                    attachmentSchema: {
+                      'ui:title': 'Document type'
+                    },
+                    attachmentName: {
```

Comment:
I don't mind doing this for now as long as it doesn't break other file fields, but we'll need to refactor this to make it more generic. I don't want to keep adding options to FileField each time a form needs more per-file fields.

Reply:
Thanks @jbalboni, sounds good.